src/web/client/common/fileSystemProvider.ts

```diff
@@ -96,10 +96,20 @@ export class PortalsFS implements vscode.FileSystemProvider {
             const castedError = error as vscode.FileSystemError;
 
             if (castedError.code === vscode.FileSystemError.FileNotFound.name) {
-                await this._loadFileFromDataverseToVFS(uri);
-
-                const data = await this._lookupAsFile(uri, false);
-                return data.data;
+                const rootDirectory = PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext().rootDirectory;
+
+                if (rootDirectory
+                    && uri.toString().includes(rootDirectory.toString())) {
+                    if (PathHasEntityFolderName(uri.toString())) {
+                        await this._loadFromDataverseToVFS();
+
+                        if (PowerPlatformExtensionContextManager.getPowerPlatformExtensionContext().defaultFileUri !== uri) {
```

Comment:
what is this for?

Reply:
now this is not needed since we have shifted to entitiy/entityid path